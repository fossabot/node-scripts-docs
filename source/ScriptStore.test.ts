import { ScriptStore } from "./ScriptStore";

it("reliably detects existence", async () => {
  expect(await ScriptStore.exists("test/fixtures/default")).toBe(true);
  expect(await ScriptStore.exists("invalid")).toBe(false);
});

it("should load an existing store", async () => {
  const store = new ScriptStore("test/fixtures/default");
  await store.load();

  expect(store.comment).toStrictEqual("This file is generated by node-scripts-docs.");
  expect(store.globalScripts).toMatchSnapshot();
  expect(store.scripts).toMatchSnapshot();
  expect(store.version).toStrictEqual(1);
});

it("parses locator as expected", () => {
  const parsed = ScriptStore.parseScriptLocator("test:coverage→node-scripts-docs");
  expect(parsed.projectName).toStrictEqual("node-scripts-docs");
  expect(parsed.scriptName).toStrictEqual("test:coverage");
});

it("renders locator as expected", () => {
  const rendered = ScriptStore.makeScriptLocator("node-scripts-docs", "test:coverage");
  expect(rendered).toStrictEqual("test:coverage→node-scripts-docs");
});
