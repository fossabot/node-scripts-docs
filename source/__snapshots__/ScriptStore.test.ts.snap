// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should load an existing store 1`] = `
Array [
  Object {
    "isGlobal": true,
    "manifestPath": "package.json",
    "projectName": "node-scripts-docs",
    "scriptCode": "docker run --rm -it -p 8000:8000 -v \${PWD}:/docs squidfunk/mkdocs-material build --site-dir=public",
    "scriptName": "docs:build",
  },
  Object {
    "isGlobal": true,
    "manifestPath": "package.json",
    "projectName": "node-scripts-docs",
    "scriptCode": "yarn nsd-debug --include-locals",
    "scriptName": "docs:scripts",
  },
  Object {
    "isGlobal": true,
    "manifestPath": "package.json",
    "projectName": "node-scripts-docs",
    "scriptCode": "docker run --rm -it -p 8000:8000 -v \${PWD}:/docs squidfunk/mkdocs-material",
    "scriptName": "docs:serve",
  },
  Object {
    "isGlobal": true,
    "manifestPath": "package.json",
    "projectName": "node-scripts-docs",
    "scriptCode": "yarn build && npm publish",
    "scriptName": "npm:publish",
  },
  Object {
    "isGlobal": true,
    "manifestPath": "package.json",
    "projectName": "node-scripts-docs",
    "scriptCode": "yarn build && npm version patch && npm publish",
    "scriptName": "npm:publish:patch",
  },
  Object {
    "isGlobal": true,
    "manifestPath": "package.json",
    "projectName": "node-scripts-docs",
    "scriptCode": "jest --coverage",
    "scriptName": "test:coverage",
  },
]
`;

exports[`should load an existing store 2`] = `
Array [
  Object {
    "isGlobal": false,
    "manifestPath": "package.json",
    "projectName": "node-scripts-docs",
    "scriptCode": "tsc",
    "scriptName": "build",
  },
  Object {
    "isGlobal": true,
    "manifestPath": "package.json",
    "projectName": "node-scripts-docs",
    "scriptCode": "docker run --rm -it -p 8000:8000 -v \${PWD}:/docs squidfunk/mkdocs-material build --site-dir=public",
    "scriptName": "docs:build",
  },
  Object {
    "isGlobal": true,
    "manifestPath": "package.json",
    "projectName": "node-scripts-docs",
    "scriptCode": "yarn nsd-debug --include-locals",
    "scriptName": "docs:scripts",
  },
  Object {
    "isGlobal": true,
    "manifestPath": "package.json",
    "projectName": "node-scripts-docs",
    "scriptCode": "docker run --rm -it -p 8000:8000 -v \${PWD}:/docs squidfunk/mkdocs-material",
    "scriptName": "docs:serve",
  },
  Object {
    "isGlobal": true,
    "manifestPath": "package.json",
    "projectName": "node-scripts-docs",
    "scriptCode": "yarn build && npm publish",
    "scriptName": "npm:publish",
  },
  Object {
    "isGlobal": true,
    "manifestPath": "package.json",
    "projectName": "node-scripts-docs",
    "scriptCode": "yarn build && npm version patch && npm publish",
    "scriptName": "npm:publish:patch",
  },
  Object {
    "isGlobal": false,
    "manifestPath": "package.json",
    "projectName": "node-scripts-docs",
    "scriptCode": "yarn build && node output/node-scripts-docs.js",
    "scriptName": "nsd-debug",
  },
  Object {
    "isGlobal": false,
    "manifestPath": "package.json",
    "projectName": "node-scripts-docs",
    "scriptCode": "jest",
    "scriptName": "test",
  },
  Object {
    "isGlobal": true,
    "manifestPath": "package.json",
    "projectName": "node-scripts-docs",
    "scriptCode": "jest --coverage",
    "scriptName": "test:coverage",
  },
]
`;
